lib/src/main/java/com/basistheory/android/view/TextElement.kt

```diff
@@ -68,15 +89,33 @@ open class TextElement @JvmOverloads constructor(
         subscribeToInputEvents()
     }
 
-    // this MUST be internal to prevent host apps from accessing the raw input values
+    // the following getters MUST be internal to prevent host apps from accessing the raw input values
+
     internal fun getText(): String? =
+        editText.text?.toString()
+
+    internal fun getTransformedText(): String? =
         editText.text?.toString().let {
             transform?.apply(it) ?: it
         }
 
     fun setText(value: String?) =
         editText.setText(value)
 
+    fun setValueRef(element: TextElement) {
+        element.addChangeEventListener {
+            setText(element.getText())
+            editText.requestLayout()
+        }
+    }
```

Comment:
I had no use of storing the valueRef into a field so I removed it. There is a slight gap here where changing the valueRef will caused two elements that are competing to set this element's value, but that is not supported at this time. Also we don't yet support removing the reference to a value. We can iterate on this in the future if it's desired.